### DISCUSSION TITLE
Do local storage stores solve Sveltekit data leaking?

### DISCUSSION THREAD

**Discussion by geanes** [joshnuss/svelte-local-storage-store#156] [Q&A] [answered]
Forgive my ignorance, but would local storage stores be a solution for the data leaking between clients behavior when using stores with Sveltekit?

**Comment by joshnuss** [comment]
Hi @geanes,

Can you explain what you mean by "data leaking"?

**Reply by geanes** [reply]
Thanks for replying! Using global stores as they are normally used in the Svelte documentation can result in data leaking between clients when used in a SvelteKit application (with SSR) because global stores are shared. The recommended workaround for this is to set up contexts to avoid leaking. I was wondering if local storage stores avoids this because the store resides on a user’s local machine, or are they still global and a copy is sent to the user’s machine. Again, please forgive me if this is a stupid question, I am building my first application with Sveltekit and using Skeleton UI which has local storage stores included.

**Comment by joshnuss** [comment, marked as answer]
I think this store is only useful on the client-side, since it relies on browser APIs.

On the server-side it acts like a regular writable store, and is probably susceptible to the same leaks Svelte's built-in stores have.

Maybe there should be a note in the readme

**Reply by geanes** [reply]
Thank you so much for the response. This is good to know!

**Reply by geanes** [reply]
This seems to be a commonly discussed Sveltekit topic, but here is more info for anyone who reads this in the future (unrelated to local storage stores): https://dev.to/brendanmatkin/safe-sveltekit-stores-for-ssr-5a0h
